Packs/KELARaDark/Integrations/RaDark/RaDark.py

```diff
@@ -0,0 +1,600 @@
+                # Prevent execution on unsupported sub types.
+                if item_res['data'].get('sub_type', -1) not in SUPPORTED_SUB_TYPES_FOR_PURCHASE:
+                    raise Exception("Sub type not supported for purchasing!")
+
+                # Extract bot ID and incident ID.
+                incident_id = item_res['data'].get('feed_property_id', '')
+                if not incident_id:
+                    raise Exception("Item ID doesn't found!")
+                bot_id = item_res['data'].get('bot_id', '')
+                if not bot_id:
+                    raise Exception("Bot ID doesn't found!")
+
+                # Check if chat room already exists.
+                incident_res = client.incident_get_items(incident_id=incident_id)
+                if isinstance(incident_res, dict) and isinstance(incident_res.get('data', ''), dict) and \
+                        isinstance(incident_res['data'].get('chat', ''), dict) and \
+                        isinstance(incident_res['data']['chat'].get('room', ''), dict):
+                    room_id = incident_res['data']['chat']['room'].get('id', '')
+
+                # Send the action status.
+                action_res = client.action_on_item(item_id=item_id, action="request")
+                if isinstance(action_res, dict) and isinstance(action_res.get('data', ''), dict) \
+                        and action_res['data'].get('value', ''):
+
+                    # Send the chat request.
+                    message = {
+                        "text": "Hi <b>@KELA</b> , I would like to acquire further details about bot: " + bot_id,
+                        "mentionsList": mentions_list
+                    }
+                    room = {"itemId": incident_id, "itemType": "FEED_PROPERTY"}
+                    if room_id:
+                        room['id'] = room_id
+
+                    # Send message.
+                    message_res = client.message_on_incident(message=message, room=room)
+                    if isinstance(message_res, dict) and isinstance(message_res.get('data', ''), dict) \
+                            and message_res['data'].get('roomId', ''):
+                        # readable_output = 'Item marked for purchasing'
+                        readable_output = 'Bot ID (' + bot_id + ') marked for purchasing'
+                    else:
+                        raise Exception("Action failed!")
+                else:
+                    raise Exception("Action failed!")
+            else:
+                readable_output = f'No data found for item ID: {incident_id}'
+        else:
+            raise Exception("Mentions list doesn't found!")
+        return CommandResults(readable_output=readable_output)
+    except Exception as e:
+        demisto.error(traceback.format_exc())  # print the traceback
+        return_error(f'Failed to execute command on {item_id}.\nError:\n {str(e)}')
+        return None
```

Comment:
Please pay attention that if the error was raised from lines that did not assign the `item_id` variable, an error will be raised. Please consider it once again.

Reply:
Will be fixed in next commit.